packages/@aws-cdk/aws-servicecatalog/lib/constraints.ts

```diff
@@ -0,0 +1,159 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sns from '@aws-cdk/aws-sns';
+import { AcceptLanguage } from './common';
+import { IProduct } from './product';
+
+/**
+ * Properties for governance mechanisms and constraints.
+ */
+export interface ConstraintProps {
+
+  /**
+   * The language code.
+   * @default - No accept language provided
+   */
+  readonly acceptLanguage?: AcceptLanguage;
+
+  /**
+   * The description of the constraint.
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * A reference to product.
+   */
+  readonly product: IProduct;
+}
+
+/**
+ * Properties for LaunchNotificationConstraint.
+ */
+export interface EventNotificationsProps extends ConstraintProps {
+
+  /**
+   * A list of SNS Topics to notify on Stack Events.
+   */
+  readonly topics: sns.ITopic[];
+}
+
+/**
+ * Properties for LaunchRoleConstraint.
+ */
+export interface LaunchRoleProps extends ConstraintProps {
+
+  /**
+   * A reference to a Role
+   */
+  readonly role: iam.IRole;
+}
+
+/**
+ * Rule for Service Catalog Template Constraint.
+ */
+interface Rule {
+
+  /**
+   * List of assertions
+   */
+  readonly Assertions: Assertion[],
+
+  /**
+   * When to apply assertions
+   */
+  readonly RuleCondition?: any
+}
+
+/**
+ * Properties for LaunchTemplateConstraint.
+ */
+interface Assertion {
+
+  /**
+   * The cfn rules to apply to the template in valid CFN JSON
+   * https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-rules.html
+   */
+  readonly Assert: any,
+
+  /**
+   * Description message provided for rule
+   */
+  readonly AssertDescription: string
+}
+
+/**
+ * Properties for LaunchTemplateConstraint.
+ */
+export interface TemplateRule {
+
+  /**
+   * List of rules
+   */
+  readonly [name: string]: Rule
+}
+
+// /**
+//  * Properties for LaunchTemplateConstraint.
+//  */
+// export interface TemplateRules {
+
+//   /**
+//    * Provisiong rules
+//    */
+//   readonly Rules: TemplateRule
+// }
+
+/**
+ * Properties for LaunchTemplateConstraint.
+ */
+export interface ProvisioningRulesProps extends ConstraintProps {
+
+  /**
+   * The rules for the template constraint
+   * @default {}
+   */
+  readonly rules?: TemplateRule;
+}
+
+/**
+ * Properties for ResourceUpdateConstraint.
+ */
+export interface TagUpdatesProps extends ConstraintProps {
+  /**
+   * Toggle for if users should be allowed to change tags
+   * @default true
```

Comment:
What does this mean if this is `false`? Why would anyone call the `allowTagUpdates` method with this `false`?

Reply:
A user could originally create this and then change it later, there is an underlying l1 resource that facilitates both the not allowed and allowed states.

Reply:
OK. How about a different approach then? Have 2 methods, `allowTagUpdates()` and `disallowTagUpdates()`?